core/controllers/editor.py

```diff
@@ -272,8 +273,8 @@ class ExplorationHandler(EditorHandler):
     def _get_exploration_data(self, exploration_id, version=None):
         """Returns a description of the given exploration."""
         try:
-            exploration = exp_services.get_exploration_by_id(
-                exploration_id, version=version)
+            exploration = exp_services.get_exp_with_draft_applied(
```

Comment:
I've added a check to get_exp_with_draft_applied() to only apply the draft if the version is valid. As far as notifying the front end about the draft being version invalid is concerned, passing the 'is_draft_version_valid' to the frontend should do it.